pkg/controller/logstash/reconcile.go

```diff
@@ -0,0 +1,82 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package logstash
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/logstash/sset"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"github.com/pkg/errors"
+
+	logstashv1alpha1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/logstash/v1alpha1"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/reconciler"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/tracing"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/utils/k8s"
+)
+
+func reconcileStatefulSet(params Params, podTemplate corev1.PodTemplateSpec) (*reconciler.Results, logstashv1alpha1.LogstashStatus) {
+	defer tracing.Span(&params.Context)()
+	results := reconciler.NewResult(params.Context)
+
+	s := sset.New(sset.Params{
+		Name:                 logstashv1alpha1.Name(params.Logstash.Name),
+		Namespace:            params.Logstash.Namespace,
+		ServiceName:          logstashv1alpha1.DefaultServiceName(params.Logstash.Name),
```

Comment:
You are setting the default service name here but you are only creating the default service if no custom service is defined.